src/scenes/Projects/Overview/ProjectOverview.tsx

```diff
@@ -0,0 +1,154 @@
+  const classes = useStyles();
+  const { data, error } = useProjectOverviewQuery();
+  const { pathname } = useLocation();
+  const project = data?.project as
+    | TranslationProject
+    | InternshipProject
+    | undefined;
+
+  const languageEngagementSection = (project?: TranslationProject) => {
+    return (
+      <>
+        <div className={classes.container}>
+          <Typography variant="h3">
+            {project?.engagements?.total} Language Engagements
+          </Typography>
+          <IconButton
+            classes={{ root: classes.addButton }}
+            aria-label="add button"
+          >
+            <Add />
+          </IconButton>
+          <Typography variant="h4">Add Language</Typography>
+        </div>
+        {project?.engagements?.items?.map((engagement) => {
+          const languageEngagement = engagement as LanguageEngagement;
+          return languageEngagement?.language ? (
+            <LanguageEngagementListItemCard {...languageEngagement} />
+          ) : null;
+        })}
+      </>
+    );
+  };
+
+  return (
+    <div className={classes.root}>
+      {error ? (
+        <Typography variant="h4">Error fetching Project</Typography>
+      ) : (
+        <>
+          <Breadcrumbs>
+            <Breadcrumb to="/projects">Project</Breadcrumb>
+            <Breadcrumb to={pathname}>{project?.name?.value}</Breadcrumb>
+          </Breadcrumbs>
+          <Typography variant="h2">{project?.name?.value}</Typography>
+          <div className={classes.container}>
+            <input
+              accept="image/*"
+              className={classes.uploadInput}
+              id="icon-button-file"
+              type="file"
+            />
+            <label htmlFor="icon-button-file">
+              <IconButton
+                classes={{ root: classes.uploadButton }}
+                aria-label="upload picture"
+                component="span"
+              >
+                <Publish />
+              </IconButton>
+            </label>
+            <Typography variant="h4">Upload Files</Typography>
+          </div>
+          <div className={classes.chips}>
+            <Chip variant="outlined" label="Mandoria, Asia" />
+            <Chip variant="outlined" label="12/20/20 - 05/04/25" />
+            <Chip variant="outlined" label="Active" />
+          </div>
+          <div className={classes.container}>
+            <BudgetOverviewCard
+              className={classes.budgetOverviewCard}
+              budget={project?.budget?.value}
+            />
+            {/* TODO This should be file card when it and api are finished */}
+            <BudgetOverviewCard budget={project?.budget?.value} />
+          </div>
+          <CardGroup>
+            <ProjectMembersSummary members={project?.team} />
+            <PartnershipSummary partnerships={project?.partnerships} />
+          </CardGroup>
+          {languageEngagementSection(project as TranslationProject)}
```

Comment:
Having difficulty finding a better way to do this. Don't like changing type but not sure how else to do it since it can be either Translation or Internship.

Reply:
Both types of projects have an engagemnet section so I wouldn't try to typecast here. When you do need to check for specifics:
```tsx
project.__typename === 'TranslationProject'
```
It would probably be easiest to just do that again with each list item
```tsx
        {project?.engagements?.items?.map((engagement) =>
          engagement.__typename === 'LanguageEngagement' ? (
            <LanguageEngagementListItemCard {...engagement} />
          ) : engagement.__typename === 'InternshipEngagement' ? (
            <InternshipEngagementListItemCard {...engagement} />
          ) : null
        )}
```

Reply:
This is changed this to not have type casting and instead uses typeName for things that are different